### PR TITLE
Feature/week view ui

### PR DESCRIPTION
**Pull request by AlexanderKozubets**

### CODE REVIEW

File: app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt [hunk 1]
```diff
@@ -1,24 +1,168 @@
...
+                day.showReports(dayReports)
+            }
+        } else {
+            day.showEmptyMessage(quotesManager.getRandomFormattedQuote())
+
+            val user = SharedPreferencesUtil.readUser() // TODO: refactor - inject account manager instead!!!
+            val firstDayOfWeek = date.firstDayOfWeekDate()
+            val weekReportsKey = key(firstDayOfWeek)
+            subscriptions[weekReportsKey]?.unsubscribe()
+            subscriptions[weekReportsKey] = timeReportRepository.getDayReports(user.id, firstDayOfWeek)
```
Comment (vaniakerer) [inline app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt]: Is it OK to use repository in presenter without some Interactor or UseCase? 
If we skip Interactors and/or UseCases we will skip business layer.
And yes, I know that they usually looks like only wrappers :)

File: app/src/main/java/co/techmagic/hr/presentation/time_tracker/ITimeTrackerRouter.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package co.techmagic.hr.presentation.time_tracker
+
+import com.techmagic.viper.Router
+import java.util.*
+
+interface ITimeTrackerRouter : Router{
```
Comment (vaniakerer) [inline app/src/main/java/co/techmagic/hr/presentation/time_tracker/ITimeTrackerRouter.kt]: Please, add space here

File: app/src/main/java/co/techmagic/hr/data/repository/TimeReportNetworkRepository.kt [hunk 3]
```diff
@@ -15,8 +18,8 @@ class TimeReportNetworkRepository(
         return setup(apiClient.getMe())
     }
 
-    override fun getDayReports(userId: String, date: String): Observable<UserReportsResponse> {
-        return setup(apiClient.getReports(userId, date))
+    override fun getDayReports(userId: String, firstDayOfWeek: Calendar): Observable<UserReportsResponse> {
+        return setup(apiClient.getReports(userId, firstDayOfWeek.firstDayOfWeekDate().formatDate("yyyy-MM-dd")))
```
Comment (vaniakerer) [inline app/src/main/java/co/techmagic/hr/data/repository/TimeReportNetworkRepository.kt]: Can we move all dateformats patterns to constants?

File: app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt (hunk 1 above)
Comment (AlexanderKozubets) [reply app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt]: I had no intention to skip interactors it was just faster to implement logic in this way. I'll leave a comment to refactor everything in a proper way.

**Comment by AlexanderKozubets** [comment]
I'll merge this PR to make a build for QA. Code refactoring and introducing an interactor abstraction will be done in the scope of other PR.

**Pull request merged**